src/main/java/internetShop/entity/User.java

```diff
@@ -3,17 +3,68 @@
 import lombok.AllArgsConstructor;
 import lombok.Data;
 import lombok.NoArgsConstructor;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
 
-import java.util.UUID;
+import javax.persistence.Entity;
+import javax.persistence.*;
+import java.util.Collection;
+import java.util.Set;
 
 @Data
 @AllArgsConstructor
 @NoArgsConstructor
-public class User {
-    //UUID id = UUID.randomUUID();
+@Entity
+@Table(name = "user")
+public class User implements UserDetails {
 
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
     private long id;
+
     private String name;
-    private int age;
+
+    private String username;
+
+    private String password;
+
+
+    @ManyToMany(fetch = FetchType.EAGER)
+    private Set<Role> roles;
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
```

Comment:
is it needed here? you are using @Data(include @Getter/@Setter)
check the behaviour

Reply:
I should implement it, it's from UserDetails interface